atlasaction/action.go

```diff
@@ -666,6 +671,74 @@ func (a *Actions) MigrateAutoRebase(ctx context.Context) error {
 	return nil
 }
 
+// MigratePlan runs the GitHub Action for "ariga/atlas-action/migrate/plan"
+func (a *Actions) MigratePlan(ctx context.Context) error {
+	tc, err := a.GetTriggerContext(ctx)
+	if err != nil {
+		return err
+	}
+	var (
+		remote     = a.GetInputDefault("remote", "origin")
+		dirURL     = a.GetInputDefault("dir", "file://migrations")
+		currBranch = tc.Branch
+	)
+	if v, err := a.exec(ctx, "git", "--version"); err != nil {
+		return fmt.Errorf("failed to get git version: %w", err)
+	} else {
+		a.Infof("migrate plan with %s", v)
+	}
+	// Since running in detached HEAD, we need to switch to the branch.
+	if _, err := a.exec(ctx, "git", "checkout", currBranch); err != nil {
+		return fmt.Errorf("failed to checkout to the branch: %w", err)
+	}
```

Comment:
Why we need this?

Reply:
you need to checkout to the branch since by default actions run or merge commit with master (detached HEAD) not on a branch.

for `git --version` it is just data for debugging (you added this to the rebase action)

Reply:
If that true, can we delay the checkout to when we need commit anything? Because this side effect if the users run many actions on the same workflows